lib/task_listener.js

```diff
@@ -113,26 +110,51 @@ export default class TaskListener extends EventEmitter {
     return cancelListener;
   }
 
-  get availableCapacity() {
+  async availableCapacity() {
     // Note: Sometimes capacity could be zero (dynamic capacity changes based on
     // shutdown and other factors) so subtracting runningTasks could result in a
     // negative number, hence the use of at least returning 0.
-    return Math.max(this.capacity - this.runningTasks.length, 0);
+    let deviceCapacity;
+    try {
+      deviceCapacity = await this.deviceManager.getAvailableCapacity();
+    }
+    catch (e) {
+      // If device capacity ccannot be determined for device managers configured
+      // for the worker, then default to 0
+      debug('Error determining device capacity');
```

Comment:
Maybe alert with `[alert-operator]` this is pretty bad right?

Capacity zero is not the problem... the problem is that we failed to determine capacity, that shouldn't happen am I right, if so how?